code/modules/admin/admin_pda_message.dm

```diff
@@ -0,0 +1,92 @@
+///Allows an admin to send messages on PDA
+/client/proc/messagePDA()
+	set name = "PDA Message"
+	set category = "Admin.Events"
+
+	if(!holder ||!check_rights(R_ADMIN))
+		return
+
+	holder.messagePDA()
+
+///Opens up the PDA Message Panel
+/datum/admins/proc/messagePDA()
+	if(!check_rights(R_ADMIN))
+		return
+
+	if(!GLOB.TabletMessengers)
```

Comment:
What I want to know is why would `TabletMessengers` ever be null? Seems to me the better solution is `.len` directly so that we runtime if something goes wrong

Reply:
currently tabletmessengers is every single tablet in the game regardless if they have the app or not, it's really messed up. i will change that eventually